tests/capture/test_base_interpreter.py

```diff
@@ -0,0 +1,439 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule tests strategy structure for defining custom plxpr interpreters
+"""
+# pylint: disable=protected-access, too-few-public-methods
+import pytest
+
+import pennylane as qml
+
+jax = pytest.importorskip("jax")
+jnp = pytest.importorskip("jax.numpy")
+
+from pennylane.capture import PlxprInterpreter  # pylint: disable=wrong-import-position
+from pennylane.capture.primitives import (  # pylint: disable=wrong-import-position
+    adjoint_transform_prim,
+    cond_prim,
+    ctrl_transform_prim,
+    for_loop_prim,
+    qnode_prim,
+    while_loop_prim,
+)
+
+pytestmark = pytest.mark.jax
+
+
+@pytest.fixture(autouse=True)
+def enable_disable_plxpr():
+    """Enable and disable the PennyLane JAX capture context manager."""
+    qml.capture.enable()
+    yield
+    qml.capture.disable()
+
+
+class SimplifyInterpreter(PlxprInterpreter):
+
+    def interpret_operation(self, op):
+        new_op = op.simplify()
+        if new_op is op:
+            new_op = new_op._unflatten(*op._flatten())
+            # if new op isn't queued, need to requeue op.
+        return new_op
+
+
+# pylint: disable=use-implicit-booleaness-not-comparison
+def test_env_and_initialized():
+    """Test that env is initialized at the start."""
+
+    interpreter = SimplifyInterpreter()
+    assert interpreter._env == {}
+    assert interpreter._op_math_cache == {}
+
+
+def test_primitive_registrations():
+    """Test that child primitive registrations dict's are not copied and do
+    not effect PlxprInterpreeter."""
+
+    class SimplifyInterpreterLocal(PlxprInterpreter):
+
+        def interpret_operation(self, op):
+            new_op = op.simplify()
+            if new_op is op:
+                # if new op isn't queued, need to requeue op.
+                new_op = new_op._unflatten(*op._flatten())
+            return new_op
```

Comment:
Just to be sneaky ? :D 
```suggestion
    class SimplifyInterpreterLocal(SimplifyInterpreter):
        pass
```

Reply:
And then could/should check against `SimplifyInterpreter` below? :thinking: